### DISCUSSION TITLE
Allow temporary download for multiplayer beatmaps

### DISCUSSION THREAD

**Discussion by Spooghetti420** [ppy/osu#20958] [Ideas]
When you go to download a beatmap for a multiplayer lobby, the map still remains afterwards even if you aren't a huge fan. Naturally, since people's tastes are different, this is gonna happen almost anytime you play in a public room. I wondered whether we could make a setting/separate download button in-lobby that downloads the map only for as long as you're still in the room, and deletes it afterwards, that way the work of having to delete tens of beatmaps after a long session is avoided and you don't have to worry about maps wasting space, etc.

**Comment by smoogipoo** [comment]
I think this would be part of a larger effort to stream beatmap data in solo play as well.
